### ISSUE TITLE
Sometimes data not displayed

### ISSUE THREAD

**Issue by torresl** [balderdashy/sails#2878]
Hi,

Sometmes data are displayed and sometimes not.
I have these errors displayed  on the console  :    
"
warn: websocket connection invalid
warn: websocket connection invalid
Error in onDisconnect:  [TypeError: Cannot read property 'id' of undefined]
Error in onDisconnect:  [TypeError: Cannot read property 'id' of undefined]
Error in onDisconnect:  [TypeError: Cannot read property 'id' of undefined]
"
I use angular js to display data.
How can i solve the problem to be sure that data will always be displayed.

Thanks

**Comment by tjwebb** [comment]
Displayed where? Please describe exactly what you're expecting to happen, in detail

**Comment by keniosor** [comment]
I just simply want to display a list of items using angular js <ng-repeat ...>.
Sometines, the list of items is displayed. sometimes not. Also, i use sails.io.js to have realtime data rendering. Perharps, it exists better than AngularJs to do it.

**Comment by tjwebb** [comment]
Post a link to your sails project.

**Comment by leobudima** [comment]
Hm, there is something to this issue...

After updating to 0.11.1 (using sails-mongo and sails.io.js), it started happening to me too. 

For the first ~2-3 min after lifting the server, I get responses to socket requests immediately. After a while, there is simply no response:

```
verbose: Receiving incoming message from Socket.io:  { method: 'get',
  headers: {},
  data: {},
  url: 'http://localhost:1337/api/v1/couriers' }
verbose: Interpreting socket.io message as virtual request to "get http://localhost:1337/api/v1/couriers"...
silly: Handling virtual request :: Running virtual querystring parser...
silly: Handling virtual request :: Running virtual body parser...

```

So it's not a problem in the client-side code, the request reaches Sails, but something gets stuck...My code hasn't changed at all from before the version bump.

Is there anything I could do to help pinpoint the problem? 

Thanks!

**Comment by torresl** [comment]
if  you have an authentification, you have to be careful for the session,
You musn't affect req.session.id but use another variable. 
for example :     
 req.session.userid = user.id;  
I changed that so my problems of displaying data where solved.

**Comment by leobudima** [comment]
Thanks for the suggestion torresl, but it's unfortunately not the cause on my side, I'm not touching the session variables...In any case, I'm glad it's solved for you.

If anyone has suggestions on what to try in my case, it would be very welcome :) I'll need to start reverting to previous versions of the libraries, otherwise, to figure out which combination doesn't have this problem :/

**Comment by rrupesh** [comment]
I've been recently having the same issues with sails socket. I was using mongo as db and also for sessions. I then shifted to mysql for db and mongo for sessions and everything seems to be working fine. no more issues with sockets or http requests. Please check if this helps @leobudima

**Comment by mikermcneil** [comment]
Thanks for posting, @torresl.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
